bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/connector/PowermaxReaderThread.java

```diff
@@ -129,9 +129,13 @@ public void run() {
             logger.debug("Interrupted via InterruptedIOException");
         } catch (IOException e) {
             logger.debug("Reading failed: {}", e.getMessage(), e);
+            connector.handleCommunicationFailure(e.getMessage());
+        } catch (Exception e) {
+            logger.debug("Error reading or processing message: {}", e.toString(), e);
+            connector.handleCommunicationFailure(e.toString());
```

Comment:
Why not e.getMessage() ?

Reply:
This is only for unhandled non-I/O exceptions, which should never happen, and might not have a message. On a NullPointerException for example, getMessage() is null, funnily enough.

Reply:
Can you at least provide the message in case there is one ?
`e.getMessage() != null ? e.getMessage() : e.toString()`

Reply:
Done